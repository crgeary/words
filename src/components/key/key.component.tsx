import classNames from "classnames";
import { ComponentPropsWithoutRef, FC } from "react";
import { CharState } from "../../types/char-state";

export type KeyProps = ComponentPropsWithoutRef<"button"> & {
    state?: CharState;
};

export const Key: FC<KeyProps> = ({ children, className, state = CharState.Idle, ...props }) => {
    return (
        <button
            className={classNames("p-2", className, {
                "bg-green-600 text-white": CharState.Correct === state,
                "bg-yellow-600 text-white": CharState.Present === state,
                "bg-slate-400 text-slate-300": CharState.Absent === state,
                "bg-slate-600 text-white": CharState.Idle === state,
            })}
            {...props}
        >
            <span className="uppercase">{children}</span>
        </button>
    );
};
