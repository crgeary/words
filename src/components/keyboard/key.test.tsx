import { render } from "@testing-library/react";
import { Key } from "./key.component";

test("it should match snapshot", () => {
    const { container } = render(<Key>A</Key>);
    expect(container.firstChild).toMatchSnapshot();
});
