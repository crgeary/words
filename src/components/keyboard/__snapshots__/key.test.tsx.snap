// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should match snapshot 1`] = `
<button
  class="inline-flex justify-center items-center p-2 border shadow-sm rounded font-medium text-sm text-center active:shadow-inner bg-white border-violet-200 text-slate-600"
>
  <span
    class="uppercase"
  >
    A
  </span>
</button>
`;
